src/uct/ib/rc/base/rc_iface.c

```diff
@@ -524,6 +524,33 @@ static int uct_rc_iface_config_limit_value(const char *name,
      }
 }
 
+static ucs_status_t
+uct_rc_iface_init_max_rd_atomic(uct_rc_iface_t *iface,
+                                const uct_rc_iface_common_config_t *config,
+                                const uct_ib_iface_init_attr_t *init_attr)
+{
+    UCS_STRING_BUFFER_ONSTACK(strb, 8);
+
+    if (config->max_rd_atomic == UCS_ULUNITS_AUTO) {
+        iface->config.max_rd_atomic = init_attr->max_rd_atomic;
+    } else if (config->max_rd_atomic <= init_attr->max_rd_atomic) {
+        iface->config.max_rd_atomic = config->max_rd_atomic;
+    } else {
+        if (config->max_rd_atomic == UCS_ULUNITS_INF){
+            ucs_string_buffer_appendf(&strb, "inf");
+        } else {
+            ucs_string_buffer_appendf(&strb, "%lu", config->max_rd_atomic);
+        }
```

Comment:
use ucs_memunits_to_str

Reply:
it looks hacky to me, because it is supposed to add some specificator in the end (even though it is not done for small values). I'd propose to leave it as it is or add similar `ucs_ulong_to_str()` func

Reply:
ic, maybe extract some common code with ucs_memunits_to_str
in any case, IMO no need to use string_buffer here; could use ucs_snprintf_safe